Code/Framework/AzFramework/AzFramework/Process/ProcessWatcher.cpp

```diff
@@ -15,14 +15,63 @@
 
 namespace AzFramework
 {
+    namespace ProcessLauncher
+    {
+        AZStd::string ProcessLaunchInfo::GetCommandLineParametersAsString() const
+        {
+            struct CommandLineParametersVisitor
+            {
+                AZStd::string operator()(const AZStd::string& commandLine) const
+                {
+                    return commandLine;
+                }
+
+                AZStd::string operator()(const AZStd::vector<AZStd::string>& commandLineArray) const
+                {
+                    AZStd::string commandLineResult;
+
+#if AZ_TRAIT_OS_USE_WINDOWS_FILE_PATHS
+                    // When re-constructing a command line from an argument list (on windows), if an argument
+                    // is double-quoted, then the double-quotes must be escaped properly otherwise
+                    // it will be absorbed by the native argument parser and possibly evaluated as
+                    // multiple values for arguments
+                    AZStd::string_view escapedDoubleQuote = R"("\")";
+
+                    AZStd::vector<AZStd::string> preprocessed_command_array;
+                    for (const auto& commandArg : commandLineArray)
+                    {
+                        AZStd::string replacedArg = commandArg;
+                        auto replacePos = replacedArg.find_first_of('"');
+                        while (replacePos != replacedArg.npos)
+                        {
+                            replacedArg.replace(replacePos, 1, escapedDoubleQuote.data(), escapedDoubleQuote.size());
+                            replacePos += escapedDoubleQuote.size();
+                            replacePos = replacedArg.find('"', replacePos);
+                        }
+                        preprocessed_command_array.emplace_back(replacedArg);
+                    }
```

Comment:
You could take the [QuoteArgument](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Settings/CommandLine.cpp#L29-L42) function for CommandLine.cpp and adapt it here.

Reply:
Not really, because the argument can be something like
```
arg1="value1"
```

Reply:
Yeah that is a flaw in my suggestion that the AZ::CommandLine doesn't have.
The AZ::CommandLine splits arguments on spaces, so this never comes up as an issue in that case.

Though what could be done is use [StringFunc::Replace](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/StringFunc/StringFunc.h#L199-L200)to escape the double quotes in one statement then.

````
auto QuoteCommandArg = [](AZStd::string_view commandArg)
{
    AZ::StringFunc::Replace(R"(")", R"(\")");
};
AZStd::for_each(commandLineArray.begin(), commandLineArray.end(), QuoteCommandArg);
```